Add buffering for resources

MichaelClifford opened the pull request:
This PR adds a new method `wait_for_resources()' that prevents the data from being published until we have values for all 64 resources. 

Without this, the weirdness score seems to grow artificially rapidly over the the first 64 entries.

Commit:
add buffering for resources

durandom commented on flatliners/comparisonscore.py:
```diff
@@ -25,7 +25,9 @@ def on_next(self, x):
             cluster_id = x['cluster']
             self.compute_cluster_distance(x)
-            self.publish(self.score[cluster_id])
+            if self.wait_for_resources(x):
```
on naming:
I would expect `wait_for_resources` to be true, if I should wait for the resources. But it's the other way round.
You could either change to `if not self.wait_for_resources`
Or, rename the method, because "positive" is always better.
`if self.ready_to_publish(x)`

MichaelClifford replied:
Ok, will change name to `ready_to_publish()`

durandom commented on flatliners/comparisonscore.py:
```diff
@@ -51,4 +53,11 @@ def compute_cluster_distance(self, values):
+    
+     def wait_for_resources(self, x):
+        cluster_id = x['cluster']
+        if len(self.clusters[cluster_id].keys()) == 63:
```
what happens, if for some reason the number of keys jump from 62 to 64? You'll never return. For checks like this, it's generally safer to check for `>= 63`
Also, at least a comment why it's 63?

MichaelClifford replied:
@durandom, working knowledge of this particular data set has shown that there are 64 resources tracked by `etcd_objects resources`. So this number shouldn't jump without someone one knowing about it. But I agree, a hard coded value is probably not the best approach here.

The main reason for this method is to simply prevent the "weirdness score" from behaving erratically, while the first round of data points are populating the initial array. perhaps a better solution would be to check if a value is being added, in which case we will not publish, or if a value is being updated, in which case we will publish. This should achieve the same goal, independent of the number of features.

durandom replied:
I'm sure this number will jump from version to version. It's arbitrary of what is being tracked in etcd.

durandom reviewed (changes requested):
some nitpicks on naming

MichaelClifford commented:
Thanks @durandom, I'll make these changes this morning.

MichaelClifford committed:
Merge branch 'master' into buffer

MichaelClifford committed:
indentation fix

MichaelClifford commented:
@durandom,  Requested changes made. : ) 

Renamed `wait_for_resources` to `ready_to-publish`. And rewrote the logic so that it is independent of a hard-coded value for feature  vector length.  

Please let me know if there is anything else before merging.

durandom reviewed (approved):
much better. thanks
